internal/clients/sts/sts_client.go

```diff
@@ -0,0 +1,36 @@
+package sts
+
+import (
+	"context"
+
+	"github.com/RHEnVision/provisioning-backend/internal/config"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	c "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/aws/aws-sdk-go-v2/service/sts/types"
+)
+
+func AssumeRole(arn string) (*types.Credentials, error) {
+	stsClient, err := CreateSTSClient()
+	if err != nil {
+		return nil, err
+	}
+
+	output, err := stsClient.AssumeRole(context.TODO(), &sts.AssumeRoleInput{
+		RoleArn:         aws.String(arn),
+		RoleSessionName: aws.String("name"),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return output.Credentials, nil
+}
```

Comment:
Please create a client type (struct) with sts.Client and context.Context fields and make it a reciever of AssumeRole. This way, a single client can be used multiple times within a request. Basically the template can be ec2 which is already in main:

```go
type Client struct {
	sts     *sts.Client
	ctx     context.Context
	log     zerolog.Logger
}

func NewSTSClient(ctx context.Context) *Client {
    // see internal/clients/ec2 for an example
}

func (c *Client) AssumeRole(arn string) (*types.Credentials, error) {
    // ... now you have c.sts, c.ctx and c.log to work with
}
```